internal/stack/_static/docker-compose-stack.yml.tmpl

```diff
@@ -170,11 +170,12 @@ services:
       interval: 60s
       timeout: 50s
       retries: 5
-    command: bash -c 'if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf'
+    # config copy is intentional that mounted volumes will be busy and cannot be overwritten
+    command: bash -c 'yes | cp /usr/share/logstash/pipeline/generated_logstash.conf /usr/share/logstash/pipeline/logstash.conf && if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf --config.reload.automatic'
```

Comment:
I think as @jsoriano mentioned this started to bloat now. May be this needs to be extracted to a script

Reply:
I had separated it but as I am not familiar with this GO project structure, I have no idea how to copy the script to container. Usually we define in `Dockerfile`, this project seems managin in `internal/compose/compose.go`?

I will add it if you let me know the entry point.

```diff
--- a/internal/stack/_static/docker-compose-stack.yml.tmpl
+++ b/internal/stack/_static/docker-compose-stack.yml.tmpl
@@ -170,8 +170,7 @@ services:
       interval: 60s
       timeout: 50s
       retries: 5
-    # config copy is intentional that mounted volumes will be busy and cannot be overwritten
-    command: bash -c 'yes | cp /usr/share/logstash/pipeline/generated_logstash.conf /usr/share/logstash/pipeline/logstash.conf && if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf --config.reload.automatic'
+    command: bash -c 'startup.sh'
     volumes:
       - "../certs/logstash:/usr/share/logstash/config/certs"
       - "../certs/elasticsearch/cert.pem:/usr/share/logstash/config/certs/elasticsearch.pem"
diff --git a/internal/stack/_static/serverless-docker-compose.yml.tmpl b/internal/stack/_static/serverless-docker-compose.yml.tmpl
index 38aba4d..e756ede 100644
--- a/internal/stack/_static/serverless-docker-compose.yml.tmpl
+++ b/internal/stack/_static/serverless-docker-compose.yml.tmpl
@@ -35,11 +35,7 @@ services:
       interval: 60s
       timeout: 50s
       retries: 5
-    # logstash expects the key in pkcs8 format. Hence converting the key.pem to pkcs8 format using openssl.
-    # Also logstash-filter-elastic_integration plugin is installed by default to run ingest pipelines in logstash.
-    # elastic-package#1637 made improvements to enable logstash stats through port 9600.
-    # config copy is intentional that mounted volumes will be busy and cannot be overwritten
-    command: bash -c 'yes | cp /usr/share/logstash/pipeline/generated_logstash.conf /usr/share/logstash/pipeline/logstash.conf && openssl pkcs8 -inform PEM -in /usr/share/logstash/config/certs/key.pem -topk8 -nocrypt -outform PEM -out /usr/share/logstash/config/certs/logstash.pkcs8.key && chmod 777 /usr/share/logstash/config/certs/logstash.pkcs8.key && if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf --config.reload.automatic'
+    command: bash -c 'startup.sh'
     volumes:
:

```

- New Logstash `startup.sh`

```#!/bin/bash
set -euo pipefail

LOGSTASH_HOME="/usr/share/logstash/"

# logstash expects the key in pkcs8 format.
# Hence converting the key.pem to pkcs8 format using openssl.
creat_pem_cert() {
  ls_cert_path="$LOGSTASH_HOME/config/certs"
  openssl pkcs8 -inform PEM -in ls_cert_path/key.pem -topk8 -nocrypt -outform PEM -out $ls_cert_path/logstash.pkcs8.key
  chmod 777 $ls_cert_path/logstash.pkcs8.key
}

# config copy is intentional that mounted volumes will be busy and cannot be overwritten
overwrite_ls_config() {
  ls_pipeline_config_path="$LOGSTASH_HOME/pipeline/"
  echo $ls_pipeline_config_path/generated_logstash.conf > $ls_pipeline_config_path/logstash.conf
}

# installs the `elastic_integration` plugin if not bundled
install_pipeline_if_missing() {
  if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then
    echo "Missing plugin logstash-filter-elastic_integration, installing now"
    bin/logstash-plugin install logstash-filter-elastic_integration;
  fi
}

# runs Logstash
run_logstash() {
  bin/logstash -f /usr/share/logstash/pipeline/logstash.conf --config.reload.automatic
}

creat_pem_cert
overwrite_ls_config
install_pipeline_if_missing
run_logstash
```

Reply:
I think you can try this

```diff
--- a/internal/stack/_static/docker-compose-stack.yml.tmpl
+++ b/internal/stack/_static/docker-compose-stack.yml.tmpl
@@ -170,8 +170,9 @@ services:
       interval: 60s
       timeout: 50s
       retries: 5
-    command: bash -c 'if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf'
+    entrypoint: bash -c 'chmod +x /usr/share/logstash/pipeline/logstash_startup.sh && /usr/share/logstash/pipeline/logstash_startup.sh'
     volumes:
+      - "./logstash_startup.sh:/usr/share/logstash/pipeline/logstash_startup.sh"
       - "../certs/logstash:/usr/share/logstash/config/certs"
       - "../certs/elasticsearch/cert.pem:/usr/share/logstash/config/certs/elasticsearch.pem"
       - "./logstash.conf:/usr/share/logstash/pipeline/logstash.conf:ro"
diff --git a/internal/stack/resources.go b/internal/stack/resources.go
index be65c76..9b4afaf 100644
--- a/internal/stack/resources.go
+++ b/internal/stack/resources.go
@@ -115,6 +115,11 @@ var (
                        Path:    ElasticAgentEnvFile,
                        Content: staticSource.Template("_static/elastic-agent.env.tmpl"),
                },
+               &resource.File{
+                       Path:         "logstash_startup.sh",
+                       CreateParent: true,
+                       Content:      staticSource.Template("_static/logstash_startup.sh"),
+               },
        }
 )
diff --git a/internal/stack/serverlessresources.go b/internal/stack/serverlessresources.go
index 8eb9b18..641da5c 100644
--- a/internal/stack/serverlessresources.go
+++ b/internal/stack/serverlessresources.go
@@ -32,6 +32,11 @@ var (
                        Path:    LogstashConfigFile,
                        Content: staticSource.Template("_static/serverless-logstash.conf.tmpl"),
                },
+               &resource.File{
+                       Path:         "logstash_startup.sh",
+                       CreateParent: true,
+                       Content:      staticSource.Template("_static/logstash_startup.sh"),
+               },
        }
 )
```

And create file `logstash_startup.sh` under  `inernal/stack/_static` folder

cc. @jsoriano @mrodm Please comment if this is OK?

Reply:
Exactly, this would need to be added as a new resource.

Reply:
Thanks @bhapas for the guide. I have applied and tested.
One thing we need to stick with is `command` instead `entrypoint`, otherwise `xpack.monitoring.enabled=false` env variable will not be in affect, Logstash internally initializes monitoring thread.

Reply:
Sure if that works